openapi/openapi/src/main/scala/endpoints/openapi/JsonSchemas.scala

```diff
@@ -119,74 +170,79 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas { openapiJsonSc
   def emptyRecord: Record[Unit] =
-    new Record(ujsonSchemas.emptyRecord, DocumentedRecord(Nil))
+    new Record(ujsonSchemas.emptyRecord, StrictDocumentedRecord(Nil))
 
   def field[A](name: String, docs: Documentation)(implicit tpe: JsonSchema[A]): Record[A] =
     new Record(
       ujsonSchemas.field(name, docs)(tpe.ujsonSchema),
-      DocumentedRecord(Field(name, tpe.docs, isOptional = false, docs) :: Nil)
+      StrictDocumentedRecord(Field(name, tpe.docs, isOptional = false, docs) :: Nil)
     )
 
   def optField[A](name: String, docs: Documentation)(implicit tpe: JsonSchema[A]): Record[Option[A]] =
     new Record(
       ujsonSchemas.optField(name, docs)(tpe.ujsonSchema),
-      DocumentedRecord(Field(name, tpe.docs, isOptional = true, docs) :: Nil)
+      StrictDocumentedRecord(Field(name, tpe.docs, isOptional = true, docs) :: Nil)
     )
 
   def taggedRecord[A](recordA: Record[A], tag: String): Tagged[A] =
     new Tagged(
       ujsonSchemas.taggedRecord(recordA.ujsonSchema, tag),
-      DocumentedCoProd(List(tag -> recordA.docs))
+      StrictDocumentedCoProd(List(tag -> recordA.docs))
     )
 
   def withDiscriminatorTagged[A](tagged: Tagged[A], discriminatorName: String): Tagged[A] =
     new Tagged(
       ujsonSchemas.withDiscriminatorTagged(tagged.ujsonSchema, discriminatorName),
-      tagged.docs.copy(discriminatorName = discriminatorName)
+      tagged.docs.withDiscriminatorName(discriminatorName)
     )
 
   def choiceTagged[A, B](taggedA: Tagged[A], taggedB: Tagged[B]): Tagged[Either[A, B]] =
     new Tagged(
       ujsonSchemas.choiceTagged(taggedA.ujsonSchema, taggedB.ujsonSchema),
-      DocumentedCoProd(taggedA.docs.alternatives ++ taggedB.docs.alternatives)
+      StrictDocumentedCoProd(taggedA.docs.alternatives ++ taggedB.docs.alternatives)
     )
 
   def zipRecords[A, B](recordA: Record[A], recordB: Record[B])(implicit t: Tupler[A, B]): Record[t.Out] =
     new Record(
       ujsonSchemas.zipRecords(recordA.ujsonSchema, recordB.ujsonSchema),
-      DocumentedRecord(recordA.docs.fields ++ recordB.docs.fields)
+      StrictDocumentedRecord(recordA.docs.fields ++ recordB.docs.fields)
     )
 
   def withExampleJsonSchema[A](schema: JsonSchema[A], example: A): JsonSchema[A] = {
     val exampleJson = schema.ujsonSchema.codec.encode(example)
     def updatedDocs(djs: DocumentedJsonSchema): DocumentedJsonSchema =
       djs match {
-        case s: DocumentedRecord => s.copy(example = Some(exampleJson))
-        case s: DocumentedCoProd => s.copy(example = Some(exampleJson))
+        case s: StrictDocumentedRecord => s.copy(example = Some(exampleJson))
+        case s: StrictDocumentedCoProd => s.copy(example = Some(exampleJson))
+        case s: LazyDocumentedRecord => new LazyDocumentedRecord(s.value.copy(example = Some(exampleJson)))
+        case s: LazyDocumentedCoProd => new LazyDocumentedCoProd(s.value.copy(example = Some(exampleJson)))
+        case s: LazySchema       => new LazySchema(updatedDocs(s.value))
```

Comment:
I initially removed LazySchema, however I moved it back (line 111) since it actually might still be useful when you want to abstract over what LazySchema you want. (We actually have an extension on this that still relies on LazySchema).

The implementation changed though, I made it a class instead of a sealed trait, so it is just three lines of code.

Reply:
However, now that I think about it, if you use the `toStrict` you don't need these anymore. Should I do that?

Reply:
Yes, at least going through `toStrict` would make it explicit that we are forcing the evaluation of the underlying schema.